src/charm.py

```diff
@@ -273,10 +273,13 @@ def _mimir_config(self) -> dict:
                     "replication_factor": 1,
                 }
             },
+            # "ruler": {
+            #     "rule_path": RULER_DATA_DIR,
```

Comment:
@rbarry82 any insight into how `ruler.rule_path` differs from what we currently use?
It seems like this one is deprecated..?

Reply:
It's not deprecated. They have different functionality.

`rule_path` is a tmpdir. When rules are created via the API or destructured, it spits them out onto disk. The [docs on it](https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/ruler/) are pretty good, with some baseline knowledge. Essentially, in sharded mode (or otherwise), rules are delegated/spread across instances through their ring buffer. When a ruler in the ring is chosen for a rule, it's written out, evaluated, and written back (either firing alerts or writing the calculated recording rule data via the API).

My real question is, to a point, "can multiple storage types be used in `ruler_storage`"? What we'd probably _like_ is the ability to _read_ rules from a local path so the lib can drop them there, then _write_ them back to object storage somewhere (or whatever the `ruler_storage` is) so we have a non-API method of provisioning that's still reslient.

I haven't read that far into the codebase, though